### PR TITLE
docs: add rename_relationship request type

### PR DESCRIPTION
**Pull request by arvi3411301**
### Description

Add `rename_relationship` request type under metadata API in docs

### Affected components

- [ ] Server
- [ ] Console
- [ ] CLI
- [x] Docs
- [ ] Community Content
- [ ] Build System
- [ ] Tests
- [ ] Other (list it)

#### Catalog upgrade

Does this PR change Hasura Catalog version?
- [ ] No
- [ ] Yes
  - [ ] Updated docs with SQL for downgrading the catalog 

#### Metadata

Does this PR add a new Metadata feature?
- [ ] No
- [ ] Yes
  - Does `run_sql` auto manages the new metadata through schema diffing?
    - [ ] Yes
    - [ ] Not required
  - Does `run_sql` auto manages the definitions of metadata on renaming?
    - [ ] Yes
    - [ ] Not required
  - Does `export_metadata`/`replace_metadata` supports the new metadata added?
    - [ ] Yes
    - [ ] Not required

#### GraphQL
- [ ] No new GraphQL schema is generated
- [ ] New GraphQL schema is being generated:
   - [ ] New types and typenames are correlated
   
   

#### Breaking changes

- [ ] No Breaking changes
- [ ] There are breaking changes:

  1. Metadata API

     Existing `query` types:
     - [ ] Modify `args` payload which is not backward compatible
     - [ ] Behavioural change of the API
     - [ ] Change in response `JSON` schema
     - [ ] Change in error code
     

  2. GraphQL API

     Schema Generation:
     
     
     - [ ] Change in any `NamedType`
     - [ ] Change in table field names
     

     Schema Resolve:-
     
     - [ ] Change in treatment of `null` value for any input fields 
     

  3. Logging

     - [ ] Log `JSON` schema has changed
     - [ ] Log `type` names have changed

### CODE REVIEW

File: docs/graphql/manual/api-reference/schema-metadata-api/relationship.rst [hunk 1]
```diff
@@ -453,3 +453,52 @@ Args syntax
      - false
      - Text
      - Comment
+
+.. _rename_relationship:
+
+rename_relationship
+------------------------
```
Comment (marionschleifer) [inline docs/graphql/manual/api-reference/schema-metadata-api/relationship.rst]:
```suggestion
-------------------
```

File: docs/graphql/manual/api-reference/schema-metadata-api/relationship.rst [hunk 2]
```diff
@@ -453,3 +453,52 @@ Args syntax
...
+
+``rename_relationship`` is used to modify name of an existing relationship.
```
Comment (marionschleifer) [inline docs/graphql/manual/api-reference/schema-metadata-api/relationship.rst]:
```suggestion
``rename_relationship`` is used to modify the name of an existing relationship.
```

File: docs/graphql/manual/api-reference/schema-metadata-api/relationship.rst [hunk 3]
```diff
@@ -453,3 +453,52 @@ Args syntax
      - false
      - Text
      - Comment
+
+.. _rename_relationship:
+
+rename_relationship
+-------------------
+
+``rename_relationship`` is used to the modify name of an existing relationship.
```
Comment (marionschleifer) [inline docs/graphql/manual/api-reference/schema-metadata-api/relationship.rst]:
```suggestion
``rename_relationship`` is used to modify the name of an existing relationship.
```

Comment (marionschleifer) [reply docs/graphql/manual/api-reference/schema-metadata-api/relationship.rst]: This was my original suggestion 🙂

File: docs/graphql/manual/api-reference/schema-metadata-api/relationship.rst [hunk 4]
```diff
@@ -453,3 +453,52 @@ Args syntax
...
+
+   * - Key
+     - Required
+     - Schema
+     - Description
+   * - table
+     - true
+     - :ref:`TableName`
+     - Name of the table
+   * - name
```
Comment (marionschleifer) [inline docs/graphql/manual/api-reference/schema-metadata-api/relationship.rst]: We just approved a PR where the `name` field here was changed to `relationship`: https://github.com/hasura/graphql-engine/pull/3419. I think this needs to be changed here too. Correct, @rakeshkky ?

Comment (rakeshkky) [reply docs/graphql/manual/api-reference/schema-metadata-api/relationship.rst]: No @marionschleifer, This is renaming a relationship. So we have `name` and `new_name` fields.

Comment (marionschleifer) [reply docs/graphql/manual/api-reference/schema-metadata-api/relationship.rst]: Okay thanks for letting me know @rakeshkky 👍

**Comment by rikinsk** [comment]
@marionschleifer Gentle reminder for the review

**Pull request merged**